### PR TITLE
Add ArrayList tests

### PR DESCRIPTION
**Pull request by YingP99**

### CODE REVIEW

File: src/System.Collections.NonGeneric/tests/ArrayList/PropertyIsSynchronizedTests.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+            alst1 = new ArrayList();
+            _alst2 = ArrayList.Synchronized(alst1);
+
+            workers = new Task[iNumberOfWorkers];
+            ts1 = new Action(AddElements);
+            // LongRunning will have problem when run in xUnit
+            for (int i = 0; i < workers.Length; i++)
+            {
+                workers[i] = Task.Factory.StartNew(ts1, TaskCreationOptions.LongRunning);
```
Comment (stephentoub) [inline src/System.Collections.NonGeneric/tests/ArrayList/PropertyIsSynchronizedTests.cs]: This should be:
Task.Factory.StartNew(ts1, CancellationToken.None, TaskCreationOptions.LongRunning, TaskScheduler.Default);
assuming you actually want it to have its own dedicated thread.  If not, you'd just make it:
Task.Run(ts1);

File: src/System.Collections.NonGeneric/tests/ArrayList/PropertyIsSynchronizedTests.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                Assert.Equal(iNumberOfTimes, iNumberOfWorkers);
+            }
+
+            //I dont think that we can make an assumption on the order of these items though
+            //now we are going to remove all of these
+            workers = new Task[iNumberOfWorkers];
+            ts1 = new Action(RemoveElements);
+            for (int i = 0; i < workers.Length; i++)
+            {
+                workers[i] = Task.Factory.StartNew(ts1, TaskCreationOptions.LongRunning);
```
Comment (stephentoub) [inline src/System.Collections.NonGeneric/tests/ArrayList/PropertyIsSynchronizedTests.cs]: Ditto.

File: src/System.Collections.NonGeneric/tests/ArrayList/PropertySyncRootTests.cs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            //we are going to rumble with the ArrayLists with 2 threads
+
+            workers = new Task[iNumberOfWorkers];
+            ts1 = new Action(SortElements);
+            ts2 = new Action(ReverseElements);
+            for (int iThreads = 0; iThreads < iNumberOfWorkers; iThreads += 2)
+            {
+
+                workers[iThreads] = Task.Factory.StartNew(ts1, TaskCreationOptions.LongRunning);
+                workers[iThreads + 1] = Task.Factory.StartNew(ts2, TaskCreationOptions.LongRunning);
```
Comment (stephentoub) [inline src/System.Collections.NonGeneric/tests/ArrayList/PropertySyncRootTests.cs]: Ditto and ditto. But I also generally question the use of LongRunning here... iNumberOfWorkers is 100... you want a hundred threads spun up?  Is that necessary for the test?

File: src/System.Collections.NonGeneric/tests/ArrayList/SynchronizedTests.cs [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+
+            alst1 = new ArrayList();
+            _ilst1 = ArrayList.Synchronized((IList)alst1);
+
+            workers = new Task[iNumberOfWorkers];
+            ts1 = new Action(AddElements);
+            // LongRunning will have problem when running under xUint
+            for (int i = 0; i < workers.Length; i++)
+            {
+                workers[i] = Task.Factory.StartNew(ts1, TaskCreationOptions.LongRunning);
```
Comment (stephentoub) [inline src/System.Collections.NonGeneric/tests/ArrayList/SynchronizedTests.cs]: Ditto.

File: src/System.Collections.NonGeneric/tests/ArrayList/SynchronizedTests.cs [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+
+                Assert.Equal(iNumberOfTimes, iNumberOfWorkers);
+            }
+
+            workers = new Task[iNumberOfWorkers];
+            ts1 = new Action(RemoveElements);
+            // LongRunning will have problem with xUnit run
+            for (int i = 0; i < workers.Length; i++)
+            {
+                workers[i] = Task.Factory.StartNew(ts1, TaskCreationOptions.LongRunning);
```
Comment (stephentoub) [inline src/System.Collections.NonGeneric/tests/ArrayList/SynchronizedTests.cs]: Ditto.

File: src/System.Collections.NonGeneric/tests/ArrayList/SynchronizedTests.cs [hunk 6]
```diff
@@ -0,0 +1,373 @@
...
+            hash = Hashtable.Synchronized(hash);
+
+            Task[] workers = new Task[7];
+
+            // initialize the threads
+            for (int i = 0; i < workers.Length; i++)
+            {
+                var name = "ThreadID " + i.ToString();
+                Action delegStartMethod = () => AddElems(name);
+                workers[i] = Task.Factory.StartNew(delegStartMethod, TaskCreationOptions.LongRunning);
```
Comment (stephentoub) [inline src/System.Collections.NonGeneric/tests/ArrayList/SynchronizedTests.cs]: Ditto.

File: src/System.Collections.NonGeneric/tests/ArrayList/PropertySyncRootTests.cs (hunk 3 above)
Comment (YingP99) [inline src/System.Collections.NonGeneric/tests/ArrayList/PropertySyncRootTests.cs]: I've changed some thread numbers, miss this one. I'll fix it.

**Comment by YingP99** [comment]
test this please

**Comment by YingP99** [comment]
@dotnet-bot add to whitelist

**Comment by weshaggard** [comment]
@dotnet-bot test this please

**Comment by YingP99** [comment]
test this please

**Pull request merged**